sqlalchemy_api_handler/bases/populate.py

```diff
@@ -121,6 +121,51 @@ def _try_to_set_attribute_with_decimal_value(self, col, key, value, expected_for
             error.add_error(col.name, "Invalid value for {} ({}): '{}'".format(key, expected_format, value))
             raise error
 
+    @classmethod
+    def _get_filter_keys(self, content, filter_keys):
+        if not isinstance(filter_keys, list):
+            filter_keys = [filter_keys]
+        return dict([(key, value) for key, value in content.items() if key in filter_keys])
+
+    @classmethod
+    def _update(self, object, content):
+        object.populate_from_dict(content)
+        return object
+
+    @classmethod
+    def _create(self, content):
+        return self(**content)
+
+    @classmethod
+    def find(self, content, filter_keys):
+        filters = self._get_filter_keys(content, filter_keys)
+        print(filters)
+        existing = self.query.filter_by(**filters).first()
+        if not existing:
+            return None
+        return existing
+
+    @classmethod
+    def find_or_create(self, content, filter_keys):
+        existing = self.find(content, filter_keys)
+        if existing:
+            return existing
+        return self._create(content)
+
+    @classmethod
+    def find_and_update(self, content, filter_keys):
+        existing = self.find(content, filter_keys)
+        if not existing:
+            return None
```

Comment:
Je comprends qu'il s'agit d'avoir le même fonctionnement que pour find, mais pour le coup, je trouve piégeux que cette méthode la puisse foirer silencieusement si l'objet n'est pas trouvé. Ça mériterait une exception je trouve, quitte à la mettre aussi coté find ? @Ledoux ?

Reply:
J'admet on y a pensé au début, j'ai juste fait un choix sur le coup mais on devrait peut être y penser de mettre une erreur pour les deux.

Un soucis par contre est de si on met une erreur sur le find, pour le find_or_create et create_or_update, il faudra mettre un try, except à l'intérieur.

Reply:
En fait je me suis basé sur l'attitude du query filter qui renvoie None, si elle n'en trouve pas.

Reply:
Yes, du coup la solution intermédiaire est pas mal: 

ne mettre l'exception que dans le find_and_update, on peut le faire sous la form de l'ApiErrors:

 def find_and_update(self, content, filter_keys):
        existing = self.find(content, filter_keys)
        if not existing:
            errors = ApiErrors()
            errors.add_error('find_and_update', 'entity not found')
            raise errors